src/slash/csv.ts

```diff
@@ -0,0 +1,85 @@
+import { RESTPostAPIApplicationCommandsJSONBody } from "discord-api-types/v10";
+import { AutocompleteInteraction, CacheType, ChatInputCommandInteraction, SlashCommandBuilder } from "discord.js";
+import { ManualTournament } from "../database/orm";
+import { AutocompletableCommand } from "../SlashCommand";
+import { getLogger, Logger } from "../util/logger";
+import { authenticateHost, autocompleteTournament, tournamentOption } from "./database";
+import * as csv from "@fast-csv/format";
+import { username } from "../util/discord";
+
+export class CsvCommand extends AutocompletableCommand {
+	#logger = getLogger("command:csv");
+
+	constructor() {
+		super();
+	}
+
+	static override get meta(): RESTPostAPIApplicationCommandsJSONBody {
+		return new SlashCommandBuilder()
+			.setName("csv")
+			.setDescription("Generate a CSV of participant details.")
+			.setDMPermission(false)
+			.setDefaultMemberPermissions(0)
+			.addStringOption(tournamentOption)
+			.addStringOption(option =>
+				option
+					.setName("type")
+					.setDescription("The type of data to display.")
+					.setRequired(true)
+					.addChoices(
+						{ name: "Pie Chart", value: "Pie Chart" },
+						{ name: "Player List", value: "Player List" }
+					)
+			)
+			.toJSON();
+	}
+
+	protected override get logger(): Logger {
+		return this.#logger;
+	}
+
+	override async autocomplete(interaction: AutocompleteInteraction<CacheType>): Promise<void> {
+		autocompleteTournament(interaction);
+	}
+
+	protected override async execute(interaction: ChatInputCommandInteraction): Promise<void> {
+		await interaction.deferReply({ ephemeral: true });
```

Comment:
CSV is ephemeral now?

Reply:
Does that affect the edited message? I was hoping it would change since `editReply` doesn't have `ephemeral`. Can we specify `ephemeral: false` there to do it? I note that the responses on failure are ephemeral.

Reply:
At the time of deferral, it must be decided whether the response is ephemeral

Reply:
https://discordjs.guide/slash-commands/response-methods.html#deferred-responses

> It is not possible to edit a reply to change its ephemeral state once sent.

Reply:
Blast. In that case we need to decide between public error messages or ephemeral response. Probably the former.